Cargo.toml

```diff
@@ -39,6 +39,7 @@ error-chain = "0.11"
 lazy_static = "0.2"
 itertools = "0.6"
 ignore = "0.2"
+sass-rs = "0.2.0"
```

Comment:
[sass-rs](https://github.com/compass-rs/sass-rs) wraps libsass.  Is there anything about that additional dependency that we should consider making this an optional feature?

Reply:
This was by far the biggest issue I ran into when I tried to initially implement this feature.

Reply:
Well, apparently, it doesn't compile on win-GNU currently :D (see my other comment). Which we should (and I believe we can) fix.

Aside from that, I don't know. As said, I am not involved in the sass thing, just want to replace my jekyll with cobalt and need sass support ;) . As such I have no knowledge about the implications this might have. But I think it might be troublesome for some embedded archs -- does cobalt support those?

How about the following: we make this a feature, but it is on by default. I think it'd be confusing to peeps, if cobalt wouldn't ship sass per default just because some archs have trouble with it. I'd rather have people explicitly disable it, if compile fails so they know it won't support sass. What do you think?

Reply:
I think being a feature is fine.  Unsure whether it should be a default one though.

Reply:
@LucioFranco 

> When I originally tried to tackle this problem, the best solution I came up with was maybe to call some extra libs, so maybe have the end user install node-sass and then call the node-sass bin from cobalt. That way we didn't not have to depend on libsass being installed externally.

Reply:
Is it dynamically or statically linking against libsass?  Our CIs pass.  I could see it maybe being installed on the Linux one, but Mac and Windows-msvc as well?

Reply:
@epage https://www.npmjs.com/package/sass here is an interesting cross-platform version of sass that might be easier for users to install (only need 3 files).

Reply:
If I read this right [libsass-sys is statically linking libsass](https://github.com/compass-rs/sass-sys/blob/master/build.rs#L75), otherwise I'd also not understand why it'd need sass-rs and sass-sys, why would you do two separate wrappers around the same dynamic lib? I understand sass-sys to be a static-linked-wrapper around libsass and sass-rs a nice rust-api on top. I might be wrong though.

>  I could see it maybe being installed on the Linux one, but Mac and Windows-msvc as well?

I think that is what sass-sys tries to provide. Just not yet for windows-gnu.

Either way, I once was told by @johannhof , that one aim of cobalt was (for him at least) to a have a one-file-all-batteries-included-executable (which is why we picked the syntax-highlighting that completely inlines all its lang-files, too) and I think that is a good goal to have. So I'd rather not want this to depend on any other build-environment or packager (like npm or ruby) to be present at execution-time (compile time, I think we can discuss). 

I'll investigate further how sass-rs and sass-sys actually do this and if not statically linked per default, I'd try to see if that is an option.

Reply:
> If I read this right sass-sys is statically linking libsass, otherwise I'd also not understand why it'd need sass-rs and sass-sys, why would you do two separate wrappers around the same dynamic lib? I understand sass-sys to be a static-linking build-environment around libsass and sass-rs a nice rust-api on top. I might be wrong though.

So it seems like it explicitly statically links on Windows but does the default build on Linux/Mac.

> I think that is what sass-sys tries to provide. Just not yet for windows-gnu.

Yeah, I didn't realize `sass-sys` built it, so I thought I could infer static linking based on what platforms might have it installed.

>  So I'd rather not want this to depend on any other build-environment or packager (like npm or ruby) to be present at execution-time (compile time, I think we can discuss).

I think the extra binary idea is meant as a work around *if* `libsass` is dynamically linked.  It effectively acts as a delay load for platforms that don't support it.  If an extra binary is needed either way, a delay load one would be nice. 

Granted, I'm more comfortable distributing for Linux with a requirement for libsass to be installed than Windows just because of the differences between the platforms (package managers).  If libsass isn't statically linked, I wonder if we could modify our CI process to get it added to our zip file.